internal/pkg/skuba/upgrade/addon/versions_test.go

```diff
@@ -28,83 +29,126 @@ import (
-					kubernetes.Kured:   &kubernetes.AddonVersion{Version: "1.2.0", ManifestVersion: 32767},
-					kubernetes.Dex:     &kubernetes.AddonVersion{Version: "2.16.0", ManifestVersion: 32767},
-					kubernetes.Gangway: &kubernetes.AddonVersion{Version: "3.1.0", ManifestVersion: 32767},
-					kubernetes.PSP:     nil,
-				},
-				Updated: kubernetes.AddonsVersion{
-					kubernetes.PSP: &kubernetes.AddonVersion{Version: "", ManifestVersion: 1},
+		test{
+			name: "skuba-config format error",
+			client: fake.NewSimpleClientset(
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      skuba.ConfigMapName,
+						Namespace: metav1.NamespaceSystem,
+					},
+					Data: map[string]string{
+						skuba.SkubaConfigurationKeyName: `
+							AddonsVersion:
+						`,
+					},
 				},
+			),
+			clusterVersion: version.MustParseSemantic("1.2.3"),
+			expectedAviu: AddonVersionInfoUpdate{
+				Current: kubernetes.AddonsVersion{},
+				Updated: kubernetes.AddonsVersion{},
 			},
-			expectedErr: false,
+			expectedErr: true,
 		},
+	)
+
+	// Test without updated
+	for _, cv := range kubernetes.AvailableVersions() {
+		client := fake.NewSimpleClientset()
+		aviu := AddonVersionInfoUpdate{
+			Current: kubernetes.AddonsVersion{},
+			Updated: kubernetes.AddonsVersion{},
+		}
+		avs := kubernetes.AllAddonVersionsForClusterVersion(cv)
+
+		// Update skuba-config configmap
+		err := skuba.UpdateSkubaConfiguration(client, &skuba.SkubaConfiguration{AddonsVersion: avs})
+		if err != nil {
+			t.Errorf("error not expected but an error was reported %v", err)
+			return
+		}
+		for addon, v := range avs {
+			aviu.Current[addon] = v
+		}
+
+		tests = append(
+			tests,
+			test{
+				name:           fmt.Sprintf("kubernetes version %s without updated", cv.String()),
+				client:         client,
+				clusterVersion: cv,
+				expectedAviu:   aviu,
+			},
+		)
+	}
+
+	// Test with updated
+	for _, cv := range kubernetes.AvailableVersions() {
+		client := fake.NewSimpleClientset()
+		aviu := AddonVersionInfoUpdate{
+			Current: kubernetes.AddonsVersion{},
+			Updated: kubernetes.AddonsVersion{},
+		}
+		avs := kubernetes.AllAddonVersionsForClusterVersion(cv)
+		for addon, v := range avs {
+			aviu.Current[addon] = nil
+			aviu.Updated[addon] = v
+		}
+
+		tests = append(
+			tests,
+			test{
+				name:           fmt.Sprintf("kubernetes version %s with updated", cv.String()),
+				client:         client,
+				clusterVersion: cv,
+				expectedAviu:   aviu,
+			},
+		)
 	}
 
 	for _, tt := range tests {
 		tt := tt // Parallel testing
 		t.Run(tt.name, func(t *testing.T) {
-			got, gotErr := UpdatedAddons(tt.clientSet, tt.clusterVersion)
-
-			if tt.expectedErr && gotErr == nil {
-				t.Errorf("error expected on %s, but no error reported", tt.name)
-				return
-			} else if !tt.expectedErr && gotErr != nil {
-				t.Errorf("error not expected on %s, but an error was reported (%v)", tt.name, gotErr)
+			gotAviu, err := UpdatedAddons(tt.client, tt.clusterVersion)
```

Comment:
nit: (fyi, I don't think nits need fixed) `aviu` is a super confusing name `addonVersionInfo` or `addonVersion` maybe? also per golint it would need to be `gotAVIU` :)

Reply:
I will keep it in mind, thanks for the suggestion :bowing_man: